pkg/nodehandler/controller.go

```diff
@@ -175,32 +182,28 @@ func (c *Controller) removeNodes(ctx context.Context, csibmnodes []nodecrd.Node)
 	return nil
 }
 
-func getTaintedNodes(nodes []corev1.Node) map[string]bool {
+func getMapIsNodesTainted(nodes []corev1.Node, taintToFind corev1.Taint) map[string]bool {
 	nodesWithTaint := map[string]bool{}
 
-	for _, node := range nodes {
-		taints := node.Spec.Taints
-		if len(taints) == 0 {
-			nodesWithTaint[node.Name] = false
-			continue
-		}
-
-		hasTaint := false
-		for _, taint := range taints {
-			if taint.Key == nodeRemovalTaintKey &&
-				taint.Value == nodeRemovalTaintValue &&
-				taint.Effect == nodeRemovalTaintEffect {
-				hasTaint = true
-				continue
-			}
-		}
-
-		nodesWithTaint[node.Name] = hasTaint
+	for i, node := range nodes {
+		nodesWithTaint[node.Name] = hasTaint(&nodes[i], taintToFind)
 	}
 
 	return nodesWithTaint
 }
 
+func hasTaint(node *corev1.Node, taintToFind corev1.Taint) bool {
+	if node == nil {
+		return false
+	}
+	for _, taint := range node.Spec.Taints {
+		if taint.Key == taintToFind.Key && taint.Value == taintToFind.Value {
```

Comment:
Why do you skip taint effect check?

Reply:
Operator doesn't check effects, and there are not rules for CSI how to use effect.

Reply:
Ok, should we delete them from consts as unusable?

Reply:
Deleted.